### PR TITLE
change cw_decred from package to plugin

### PR DESCRIPTION
**Pull request by itswisdomagain**
We'll need cw_decred to be a plugin so we can set up each OS target to use the dcrlibwallet c library built from go. Linking the library to each target will come separately.

### CODE REVIEW

File: cw_decred/ios/cw_decred.podspec [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+Pod::Spec.new do |s|
+  s.name             = 'cw_decred'
+  s.version          = '0.0.1'
+  s.summary          = 'A new Flutter plugin project.'
+  s.description      = <<-DESC
+A new Flutter plugin project.
+                       DESC
+  s.homepage         = 'http://example.com'
+  s.license          = { :file => '../LICENSE' }
+  s.author           = { 'Your Company' => 'email@example.com' }
```
Comment (itswisdomagain) [inline cw_decred/ios/cw_decred.podspec]:
```suggestion
  s.summary          = 'Cake Wallet Decred'
  s.description      = 'Cake Wallet wrapper over Decred project'
  s.homepage         = 'http://cakewallet.com'
  s.license          = { :file => '../LICENSE' }
  s.author           = { 'Cake Wallet' => 'support@cakewallet.com' }
  s.source           = { :path => '.' }
```

File: cw_decred/macos/cw_decred.podspec [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+Pod::Spec.new do |s|
+  s.name             = 'cw_decred'
+  s.version          = '0.0.1'
+  s.summary          = 'A new Flutter plugin project.'
+  s.description      = <<-DESC
+A new Flutter plugin project.
+                       DESC
+  s.homepage         = 'http://example.com'
+  s.license          = { :file => '../LICENSE' }
+  s.author           = { 'Your Company' => 'email@example.com' }
```
Comment (itswisdomagain) [inline cw_decred/macos/cw_decred.podspec]:
```suggestion
  s.summary          = 'Cake Wallet Decred'
  s.description      = 'Cake Wallet wrapper over Decred project'
  s.homepage         = 'http://cakewallet.com'
  s.license          = { :file => '../LICENSE' }
  s.author           = { 'Cake Wallet' => 'support@cakewallet.com' }
  s.source           = { :path => '.' }
```

**Review by itswisdomagain — COMMENTED** [review]
Missed these.

File: cw_decred/.metadata [hunk 3]
```diff
@@ -1,10 +1,36 @@
 # This file tracks properties of this Flutter project.
 # Used by Flutter tool to assess capabilities and perform upgrades etc.
 #
-# This file should be version controlled and should not be manually edited.
+# This file should be version controlled.
```
Comment (JoeGruffins) [inline cw_decred/.metadata:4]: Did you change these manually?

Also what is the commit below? f468f3366c26a5092eb964a230ce7892fda8f2f8

File: cw_decred/.metadata [hunk 4]
```diff
@@ -1,10 +1,36 @@
...
+
+  # User provided section
+
+  # List of Local paths (relative to this file) that should be
+  # ignored by the migrate tool.
+  #
+  # Files that are not part of the templates will be ignored by default.
+  unmanaged_files:
+    - 'lib/main.dart'
+    - 'ios/Runner.xcodeproj/project.pbxproj'
```
Comment (JoeGruffins) [inline cw_decred/.metadata:36]: These don't currently exist do they?

File: cw_decred/README.md [hunk 5]
```diff
@@ -1,3 +1,15 @@
...
+
+This project is a starting point for a Flutter
+[plug-in package](https://flutter.dev/developing-packages/),
+a specialized package that includes platform-specific implementation code for
+Android and/or iOS.
+
+For help getting started with Flutter development, view the
+[online documentation](https://flutter.dev/docs), which offers tutorials,
+samples, guidance on mobile development, and a full API reference.
+
```
Comment (JoeGruffins) [inline cw_decred/README.md:15]: Maybe revert to the TODO? This is what was there before I put the TODO message.

**Review by JoeGruffins — APPROVED** [review]
What does a plugin do? I'll google anyway.

Looks fine, do you want to pr these changes to the cake wallet repo? I think the first pr is big enough, also it's probably better that you handle any convos that arise from these changes since you have a handle on implementation.

Maybe wait to see what they say about https://github.com/cake-tech/cake_wallet/pull/1165

**Comment by itswisdomagain** [comment]
Closing in favor of #8.

**Pull request closed without merging**